upload-api/stores/blobs.js

```diff
@@ -0,0 +1,134 @@
+import { base64pad } from 'multiformats/bases/base64'
+import { decode as digestDecode } from 'multiformats/hashes/digest'
+import { base58btc } from 'multiformats/bases/base58'
+
+import {
+  S3Client,
+  HeadObjectCommand,
+  PutObjectCommand,
+} from '@aws-sdk/client-s3'
+import { getSignedUrl } from '@aws-sdk/s3-request-presigner'
+
+/**
+ * @typedef {import('@web3-storage/upload-api/types').BlobsStorage} BlobsStorage
+ * @typedef {import('@ucanto/interface').Failure} Failure
+ * @typedef {import('@ucanto/interface').Result<boolean, Failure>} HasResult
+ */
+
+/**
+ * Abstraction layer with Factory to perform operations on bucket storing Blobs.
+ *
+ * @param {string} region
+ * @param {string} bucketName
+ * @param {import('@aws-sdk/client-s3').ServiceInputTypes} [options]
+ */
+export function createBlobsStorage(region, bucketName, options) {
+  const s3 = new S3Client({
+    region,
+    ...options,
+  })
+  return useBlobsStorage(s3, bucketName)
+}
+
+/**
+ *
+ * @param {S3Client} s3
+ * @param {string} bucketName
+ * @returns {BlobsStorage}
+ */
+export function useBlobsStorage(s3, bucketName) {
```

Comment:
This is quite similar with `buckets/car-store` with few modifications given new key schema

Reply:
Might be worth adding this as code comment actually.